stable/external-dns/templates/deployment.yaml

```diff
@@ -187,6 +187,13 @@ spec:
         {{- end }}
         # Cloudflare environment variables
         {{- if eq .Values.provider "cloudflare" }}
+        {{- if .Values.cloudflare.apiToken }}
```

Comment:
This is okay, but is weird. What happen when neither the `apiKey` or `apiToken` are provided and the provider is `cloudflare`?
See the comment on the `secret.yaml`, please.

Reply:
@miguelaeh like you can read on the [documentation](https://github.com/kubernetes-incubator/external-dns/blob/master/docs/tutorials/cloudflare.md#creating-cloudflare-credentials)

> API Token will be preferred for authentication if CF_API_TOKEN environment variable is set. Otherwise CF_API_KEY and CF_API_EMAIL should be set to run ExternalDNS with Cloudflare.

wdyt?

Reply:
Ah, okay, I didn't knew that, thanks for the clarification.